src/main/subgridWeightsMod.F90

```diff
@@ -334,9 +336,17 @@ logical function is_active_l(l, glc_behavior)
           is_active_l = .true.
        end if
 
-       if ((lun%itype(l) >= isturb_MIN .and. lun%itype(l) <= isturb_MAX) .and. &
-            run_zero_weight_urban) then
-          is_active_l = .true.
+       ! Set urban land units to active, as long as memory has been allocated for such land units, either
+       ! through the run_zero_weight_urban setting which runs all urban landunits in each grid cell or
+       ! through pct_urban_max which is the maximum percent urban for each density type in a transient run.
+       ! By doing this, urban land units are also run virtually in grid cells which will grow
+       ! urban during the transient run.
+
+       if ( lun%itype(l) >= isturb_MIN .and. lun%itype(l) <= isturb_MAX ) then 
+          dens_index = lun%itype(l) - isturb_MIN + 1
+          if (run_zero_weight_urban .or. pct_urban_max(g,dens_index) > 0._r8) then
+             is_active_l = .true.
+          end if
```

Comment:
I'm not positive, but I think this could safely be changed to just return `.true.` for all urban landunits, similarly to what is done for lakes: this would say, if we have bothered to allocate memory for an urban landunit for any reason, then we want to make it active. I'm pretty sure that's what is intended here. However, the current code seems fine, too, so I'll leave it up to you whether you want to change it. The main advantages of changing it would be (1) consistency with lake, and (2) future-proofing in case the logic in subgridMod for when to allocate urban changes (currently the logic here needs to be kept consistent with the logic there).

Reply:
Given the current logic, I'm wondering why such an urban landunit would be allocated in the first place: if it has zero-weight throughout the time series, shouldn't the logic in `urban_landunit_exists` lead it to not getting allocated?

Reply:
Hmm, no, I think you are right.  In both cases an ncdump of the h1 file gives:

data:
 land1d_wtgcell = 1, 0, 0, 0 ;
 land1d_ityplunit = 1, 2, 7, 8 ;
 land1d_active = 1, _, 1, 1 ;

which is correct since the landuse.timeseries has zero percent urban for MD (land1d_ityplunit=9).

I agree that it would be good to be consistent, so I'll make the change you suggested.

Reply:
Sounds good, thanks for checking.